File: tournaments/extractMatrixColumn/extractMatrixColumn.cpp
typedef std::vector<int> vect;

vect extractMatrixColumn(std::vector<vect> matrix, int column) {

  vect result;
  for (int i = 0; i < matrix.size(); i++) {
    result.push_back(matrix[i][column]);
  }

  return result;
}

File: tournaments/extractMatrixColumn/extractMatrixColumn.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub extractMatrixColumn {
    my ($matrix, $column) = @_;

    my @result;

    for (my $i = 0; @$matrix > $i; ++$i) {
        push(@result, @{@$matrix[$i]}[$column]);
    }

    return \@result;
}
